### PR TITLE
[1t7g5t1] Create mutation startGame

### PR DESCRIPTION
**Pull request by DenisZackharov**
### Summary

Added the mutation StartGame, which returns the gameid and the question.
### How it works

Screenshots/screencasts of the pull request introduced functionality.

### Test plan

```
mutation {
  startGame {
    gameId
    question {
      avatarUrl
      answerOptions
    }
  }
}
```

### Review notes

While reviewing pull-request (especially when it's your pull-request),
please make sure that:

- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- DB schema changes reflect new migrations
- newly introduced DB fields have indexes and constraints
- there are no missed files (migrations, view templates)
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### Deploy notes

Add new ENV variable GAME_TIME_SECONDS=55.

### References
* Resolves (https://app.clickup.com/t/1t7g5t1) Issue
* [GitHub Pull Request [1t7g6t1](https://github.com/fs/facegame-api/pull/26)

### CODE REVIEW

File: app/graphql/types/start_game_type.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module Types
+  class StartGameType < Types::BaseObject
+    field :game_id, ID, null: false
+    field :question, Types::TeamMember, null: false
+  end
+
+  def game_id
+    object.id
```
Comment (KilinMikhail) [inline app/graphql/types/start_game_type.rb:8]: `object.result.id`

File: app/interactors/start_game/create_result.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    delegate :current_user, to: :context
+
+    def call
+      context.result = create_result
+      context.fail!(error_data: error_data) unless result.save
+    end
+
+    private
+
+    def create_result
```
Comment (KilinMikhail) [inline app/interactors/start_game/create_result.rb:14]: `def result`

File: app/interactors/start_game/create_result.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+class StartGame
+  class CreateResult
+    include Interactor
+
+    delegate :current_user, to: :context
+
+    def call
+      context.result = create_result
```
Comment (KilinMikhail) [inline app/interactors/start_game/create_result.rb:8]: remove this line

File: app/interactors/start_game/create_result.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    
+    def result_params
+      {
+        user: current_user,
+        finished_at: ENV.fetch("GAME_TIME", 55).to_i.second.since
+      }
+    end
+
+    def error_data
+      { message: "Result can't save" }
```
Comment (DenisZackharov) [inline app/interactors/start_game/create_result.rb:26]: Change error data message

File: app/graphql/types/start_game_type.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+module Types
+  class StartGameType < Types::BaseObject
+    field :game_id, ID, null: false
+    field :question, Types::TeamMember, null: false
```
Comment (KilinMikhail) [inline app/graphql/types/start_game_type.rb:4]: Types::TeamMemberType

File: spec/interactors/start_game/create_result_spec.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
+require "rails_helper"
+
+describe StartGame::CreateResult do
+  include_context "with interactor"
+ 
+    let(:initial_context) do
+      {
+        result: result,
+        result_params: result_params
```
Comment (KilinMikhail) [inline spec/interactors/start_game/create_result_spec.rb:9]: current_user: current_result

**Pull request merged**